### PR TITLE
[GNA]: Split eltwise using ngraph

### PR DESCRIPTION
**Pull request by nkogteva**
### Details:
 - *Move legacy EltwiseSplitOverChannelsPass to ngraph transformation*

### Tickets:
 - *52101*

### CODE REVIEW

File: src/plugins/intel_gna/gna_plugin.cpp [hunk 1]
```diff
@@ -823,9 +829,9 @@ void GNAPlugin::LoadNetwork(const CNNNetwork& _network) {
 
         if (!isNgraphPassesUsed) {
             passes->registerPass<ReorderMaxPoolPass>();
+            passes->registerPass<EltwiseSplitOverChannelsPass>();
```
[the remark this answers is not in the document]
Comment (nkogteva) [reply src/plugins/intel_gna/gna_plugin.cpp:832]: > What's the reason for still leaving this?

I'm answering this question in each PR). There is only one way how non-ngraph model can come - via legacy tests, in release it is impossible.

File: src/plugins/intel_gna/transformations/split_eltwise_over_channel.hpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace ov {
+namespace intel_gna {
+namespace pass {
+/**
+ * @brief Split over channels for Eltwise to avoid GNA-HW limitation of 65k elements per eltwise
+ */
+class SplitEltwiseOverChannel : public ov::pass::MatcherPass {
+public:
+  OPENVINO_RTTI("SplitEltwiseOverChannel", "0");
```
[the remark this answers is not in the document]
Comment (nkogteva) [reply src/plugins/intel_gna/transformations/split_eltwise_over_channel.hpp:17]: > I'm wondering if the name should include "OverChannel". From the code it looks like the split can be done on any axis.

It is a good question) I just kept old name. Maybe there is a sense to remove it

File: src/plugins/intel_gna/gna_plugin.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (sirzabek) [reply src/plugins/intel_gna/gna_plugin.cpp:832]: Maybe in two PRs ;-) I'm just reminding this way that we should start rewriting the legacy tests and removing the legacy code.

**Review by sirzabek — APPROVED** [review]

File: src/plugins/intel_gna/transformations/split_eltwise.hpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+#pragma once
+
+#include <openvino/pass/graph_rewrite.hpp>
+
+namespace ov {
+namespace intel_gna {
+namespace pass {
+/**
+ * @brief Split over channels for Eltwise to avoid GNA-HW bufferMaxSize limitation per eltwise
```
Comment (mryzhov) [inline src/plugins/intel_gna/transformations/split_eltwise.hpp:13]: Can we describe here the splitting rules?

**Pull request merged**